client/landing/gutenboarding/components/signup-form/index.tsx

```diff
@@ -158,16 +167,25 @@ const SignupForm = ( { onRequestClose }: Props ) => {
 	}
 
 	const langFragment = lang ? `/${ lang }` : '';
+	const anchorFmParameter = isAnchorFmSignup ? `&anchor_podcast=${ anchorFmPodcastId }` : '';
 	const loginRedirectUrl = encodeURIComponent(
-		`${ window.location.origin }/new${ makePath( Step.CreateSite ) }?new`
+		`${ window.location.origin }/new${
+			isAnchorFmSignup ? makePath( Step.IntentGathering ) : makePath( Step.CreateSite )
+		}?new${ anchorFmParameter }`
```

Comment:
So this could end up as `/new/path/?new`? Hmm..

Reply:
Thanks, I [pushed a change](https://github.com/Automattic/wp-calypso/pull/47600/commits/38f5e4b52de57f9f5497447622779d0ec49c5a26) so that the two login redirect URL possibilities are:
- /new/create-site?new (the current redirect path to site creation)
- /new/?new&anchor_podcast={id} (the Anchor.fm flavored redirect path to the My podcast is called... step which is the root path)